portfolio/src/main/webapp/script.js

```diff
@@ -12,49 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/** Fetches a comment count from the server.
-function getCommentCount() {
-  console.log('Fetching comment count.');
-  fetch('/data-limit').then(response => response.text()).then((commentCount) => {
-    console.log('Number of comments to load: ' + commentCount);
-  });
-  const commentsElement = document.getElementById('comments-container');
-  for (i = 0; i <  commentCount; i++) {
-      
-  }
-}
-*/
 
 /**
- * Fetches a form submission/comment from the server and adds it to DOM
+ * Fetches a form submission/comment from the server and adds it to DOM.
  */
 function getFormResponse() {
+  const commentsMaxElement = document.getElementById('count').value;
+  console.log('Number of comments to be displayed: ' + commentsMaxElement);
   console.log('Fetching responses to form.');
-  const responsePromise = fetch('/data');
-  responsePromise.then(handleResponse);
-}
-
-/**
- * Handles response by converting it to text and passing the result to
- * addQuoteToDom().
- */
-function handleResponse(response) {
-  console.log('Handling the response.');
-  const textPromise = response.text();
-  textPromise.then(addResponseToDom);
-}
-
-/** Adds form submission/comment to the DOM. */
-function addResponseToDom(comments) {
-  console.log('Adding message to dom: ' + comments);
-  const commentsElement = document.getElementById('comments-container');
-  const commentsObj = JSON.parse(comments);
-  commentsObj.forEach((comment) => {
+  fetch('/data?count=' + commentsMaxElement).then(response => response.text()).then((comments) => {
+    const commentsElement = document.getElementById('comments-container');
+    const commentsObj = JSON.parse(comments);
+    commentsElement.innerText = '';
+    commentsObj.forEach((comment) => {
       commentsElement.appendChild(createCommentElement(comment));
+    })
   });
 }
 
-/** Creates an <li> element containing form submission/comment. */
+/** 
+ * Creates an <li> element containing a form submission/comment.
+ */
 function createCommentElement(comment) {
   const commentElement = document.createElement('li');
```

Comment:
Next todo: add a comment what this parameter comment.

I would perhaps also change the naming to 

server_comment (or something distinguishable). I was at first confused where this variable was coming from in L42)